ddtrace/internal/telemetry/telemetry_writer.py

```diff
@@ -0,0 +1,234 @@
+from collections import namedtuple
+import logging
+from typing import ClassVar
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ddtrace.internal import forksafe
+
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .telemetry_request import app_closed_telemetry_request
+from .telemetry_request import app_integrations_changed_telemetry_request
+from .telemetry_request import app_started_telemetry_request
+
+
+TelemetryRequest = namedtuple("TelemetryRequest", "request fail_count")
+
+log = get_logger(__name__)
+
+
+def _get_interval_or_default():
+    return float(get_env("instrumentation_telemetry", "interval", default=60))
+
+
+class TelemetryWriter(PeriodicService):
+    """
+    Periodic service which sends Telemetry request payloads to the agent-proxy [not yet but soon]
+    """
+
+    AGENT_URL = get_trace_url()
+    ENDPOINT = "telemetry/proxy/api/v2/apmtelemetry"
+    MAX_FAIL_COUNT = 3
+
+    enabled = False  # type: ClassVar[bool]
+    _instance = None  # type: ClassVar[Optional[TelemetryWriter]]
+    _lock = forksafe.Lock()  # type: ClassVar[forksafe.ResetObject]
+    sequence = 0  # type: ClassVar[int]
+
+    def __init__(self):
+        # type: () -> None
+        super(TelemetryWriter, self).__init__(interval=_get_interval_or_default())
+
+        self.encoder = JSONEncoderV2()
+        self._events_queue = []  # type: List[TelemetryRequest]
+        self._integrations_queue = []  # type: List[Dict]
+
+    def _get_headers(self, payload_type):
+        # type: (str) -> Dict
+        """
+        Creates request headers
+        """
+        return {
+            "Content-type": "application/json",
+            "DD-Telemetry-Request-Type": payload_type,
+            "DD-Telemetry-API-Version": "v1",
```

Comment:
Why is this `v1` when the request url uses `v2`?

Reply:
fixed